### PR TITLE
[DOC] Rewrite current user installation guide and move dev installation guide to its own file

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Part of #168.

See title. The current guide is out of date for aeon.

New: https://aeon-toolkit--524.org.readthedocs.build/en/524/installation.html
Old: https://www.aeon-toolkit.org/en/latest/installation.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # ⌛ Welcome to **aeon**
 
 `aeon` is an open source toolkit for learning from time series. It is compatible with
-[scikit-learn](https://scikit-learn.org) and provides access to the very latest
+[scikit-learn](https://scikit-learn.org) and provides access to the  very latest
```
Comment (TonyBagnall) [inline README.md:8]: extra whitespace?

File: docs/developer_guide/dev_installation.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+This allows editing and extending the code in-place. See also
+`pip reference on editable installs <https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs>`_).
+
+```{note}
+    You will have to re-run:
+
+    pip install --editable .
+
+    every time the source code of a compiled extension is changed (for
+    instance when switching branches or pulling changes from upstream).
```
Comment (TonyBagnall) [inline docs/developer_guide/dev_installation.md:68]: is this true? you need to rerun it every time you switch branches? I dont think I have done so

**Review by TonyBagnall — APPROVED** [review]

File: docs/developer_guide/dev_installation.md (hunk 2 above)
Comment (MatthewMiddlehurst) [reply docs/developer_guide/dev_installation.md:68]: This is the same as the dev instructions currently in place, just in a different file.

It should be redone at some point though.

**Review by TonyBagnall — APPROVED** [review]
lgtm

**Pull request merged**